confirmation_wizard/wizard/confirmation_wizard.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from ast import literal_eval
+
+from odoo import _, api, fields, models
+
+
+class ConfirmationWizard(models.TransientModel):
+    _name = "confirmation.wizard"
+    _description = "Confirmation Wizard"
+
+    message = fields.Char(string="Confirm Message", required=True)
+
+    res_ids = fields.Char()
+    res_model = fields.Char()
+
+    callback_method = fields.Char()
+    callback_params = fields.Json()
+
+    return_type = fields.Selection(
+        [
+            ("window_close", "Return Window Close Action"),
+            ("method", "Return Method"),
+        ],
+        default="window_close",
+        required=True,
+    )
+
+    @api.model
+    def _prepare_action(self, title=None):
+        """
+        Prepare confirmation wizard
+        :param title: wizard title
+        """
+        action = self.env["ir.actions.act_window"]._for_xml_id(
+            "confirmation_wizard.confirmation_wizard_action"
+        )
+        action.update(
+            {
+                "name": title or _("Confirmation"),
+                "res_id": self.id,
+                "context": self._context,
+            }
+        )
+        return action
+
+    @api.model
+    def confirm_message(
+        self, message, records, title=None, method=None, callback_params=None
+    ):
+        """
+        Confirm message with method return type
+        :param message: confirmation message
+        :param records: record set
+        :param title: wizard title
+        :param method: triggered method
+        :param callback_params: method arguments
+        :return dict: ir actions act window dict
+        """
+        wizard = self.create(
+            {
+                "message": message,
+                "res_ids": repr(records.ids),
+                "return_type": "method",
+                "res_model": records._name,
+                "callback_method": method,
+                "callback_params": callback_params or {},
+            }
+        )
+        return wizard._prepare_action(title)
+
+    @api.model
+    def confirm_no_action_message(self, message, title=None):
+        """
+        Confirm message with close window return type
+        :param message: confirmation message
+        :param title: wizard title
+        :return dict: ir actions act window dict
```

Comment:
You should include `invisible_cancel` as a special context here and how it works. I think it should be `show_cancel` is False to hide it, default to True, or `hide_cancel` is True to hide the Cancel button. `invisible_cancel` isn't a proper word in english

Reply:
> You should include `invisible_cancel` as a special context here and how it works. I think it should be `show_cancel` is False to hide it, default to True, or `hide_cancel` is True to hide the Cancel button. `invisible_cancel` isn't a proper word in english

This context key used as special configuration and use to all methods where confirmation wizard call.

Reply:
even it's already been in the documentation, it's a good thing to put that context in this docstring as well as they are closely used together